lib/install/management/dispatcher.go

```diff
@@ -125,12 +131,60 @@ func (d *Dispatcher) initDiagnosticLogs(conf *metadata.VirtualContainerHostConfi
 	}
 }
 
+func (d *Dispatcher) registerExtension(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infoln("Registering VCH as a vSphere extension")
+	extensionManager := object.NewExtensionManager(d.session.Vim25())
+	extension := types.Extension{
+		Description: &types.Description{
+			Label:   "VIC",
+			Summary: "vSphere Integrated Containers Virtual Container Host",
+		},
+		Company: "VMware, Inc.",
+		Version: "0.2",
+		Key:     "com.vmware.vic",
+	}
+
+	var err error
+	var certbuffer, keybuffer bytes.Buffer
+	if certbuffer, keybuffer, err = certificate.CreateRawKeyPair(); err != nil {
```

Comment:
@gigawhitlocks what I mean is not for where to register the extension, which should happen after the appliance VM creation, then you can get the VM mob-id or UUID.
But for the certificate generation, cause theoretically it is still possible to have one certificate generation error, so it will be better to have the generation finished before we start VC connection.
That's why we have user (or docker) certificate generated in create package.
I assume what you're doing is a similar process.